Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,714 @@
+  DFunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub (fun _ _ => rfl)
+    fun _ _ => rfl
+
+instance instNonUnitalCommRing [NonUnitalCommRing β] [TopologicalRing β] :
+    NonUnitalCommRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instIsScalarTower {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [IsScalarTower R β β] :
+    IsScalarTower R C_c(α, β) C_c(α, β) where
+  smul_assoc r f g := by
+    ext
+    simp only [smul_eq_mul, coe_mul, coe_smul, Pi.mul_apply, Pi.smul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_assoc]
+
+instance instSMulCommClass {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [SMulCommClass R β β] :
+    SMulCommClass R C_c(α, β) C_c(α, β) where
+  smul_comm r f g := by
+    ext
+    simp only [smul_eq_mul, coe_smul, coe_mul, Pi.smul_apply, Pi.mul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_comm]
+
+end AlgebraicStructure
+
+section ZeroAtInfty
+
+open ZeroAtInfty
+
+variable [TopologicalSpace β] [TopologicalSpace γ] [Zero γ]
+variable [FunLike F β γ] [CompactlySupportedContinuousMapClass F β γ]
+
+lemma zero_at_infty_of_hasCompactSupport (f : F) :
+    Filter.Tendsto f (Filter.cocompact β) (𝓝 0) := by
+  rw [_root_.tendsto_nhds]
+  intro s _ hzero
+  rw [Filter.mem_cocompact]
+  use tsupport f
+  constructor
+  · exact hasCompactSupport f
+  · intro x hx
+    simp only [Set.mem_preimage]
+    rw [← Set.not_mem_compl_iff, compl_compl] at hx
+    rw [image_eq_zero_of_nmem_tsupport hx]
+    exact hzero
+
+instance : ZeroAtInftyContinuousMapClass C_c(β, γ) β γ where
+  map_continuous f := f.continuous_toFun
+  zero_at_infty f := zero_at_infty_of_hasCompactSupport f
+
+end ZeroAtInfty
+
+section Uniform
+
+variable [UniformSpace β] [UniformSpace γ] [Zero γ]
+variable [FunLike F β γ] [CompactlySupportedContinuousMapClass F β γ]
+
+theorem uniformContinuous (f : F) : UniformContinuous (f : β → γ) :=
+  (map_continuous f).uniformContinuous_of_tendsto_cocompact (zero_at_infty_of_hasCompactSupport f)
+
+end Uniform
+
+/-! ### Metric structure
+
+When `β` is a metric space, then every element of `C_c(α, β)` is bounded, and so there is a natural
+inclusion map `CompactlySupportedContinuousMap.toBCF : C_c(α, β) → (α →ᵇ β)`. Via this map
+`C_c(α, β)` inherits a metric as the pullback of the metric on `α →ᵇ β`. Moreover, this map has
+closed range in `α →ᵇ β` and consequently `C_c(α, β)` is a complete space whenever `β` is complete.
+-/
+
+
+section Metric
+
+open Metric Set
+
+variable [PseudoMetricSpace β] [Zero β] [FunLike F α β] [CompactlySupportedContinuousMapClass F α β]
+
+protected theorem bounded (f : F) : ∃ C, ∀ x y : α, dist ((f : α → β) x) (f y) ≤ C := by
```

Comment:
The full name of this theorem is `CompactlySupportedContinuousMap.bounded`. It's a little bit misleading, as it applies not only to `CompactlySupportedContinuousMap`, but in fact to all `CompactlySupportedContinuousMapClass`. So rather call it `CompactlySupportedContinuousMapClass.bounded`, i.e., don't put the results of this section in the `CompactlySupportedContinuousMap` namespace.

Reply:
Sorry, I don't know how to do this. Some results are about `F`, but when I move them out of `CompactlySupportedContinuousMap` and `under `CompactlySupportedContinuousMapClass`, some notations such as `f.toBCF` are no longer valid. Could you help?

Reply:
actually, in accordance with your comment below, I removed the section `Metric`, so this is no longer a problem.